daemon/daemon.go

```diff
@@ -744,6 +745,7 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 }
 
 func (daemon *Daemon) Shutdown() error {
+	daemon.shutdown = true
```

Comment:
yes, I think it could cause wrong behavior if a container is stopped at the same time the daemon is being stopped. We might need to protect that field with a lock.

@LK4D4 what do you think?